manticore/platforms/linux.py

```diff
@@ -2036,26 +2056,84 @@ class SLinux(Linux):
     Builds a symbolic extension of a Linux OS
 
     :param str programs: path to ELF binary
+    :param str disasm: disassembler to be used
     :param list argv: argv not including binary
     :param list envp: environment variables
     :param tuple[str] symbolic_files: files to consider symbolic
     """
-    def __init__(self, programs, argv=None, envp=None, symbolic_files=None):
+    def __init__(self, programs, argv=None, envp=None, symbolic_files=None,
+                 disasm='capstone'):
         argv = [] if argv is None else argv
         envp = [] if envp is None else envp
         symbolic_files = [] if symbolic_files is None else symbolic_files
 
         self._constraints = ConstraintSet()
         self.random = 0
         self.symbolic_files = symbolic_files
-        super(SLinux, self).__init__(programs, argv, envp)
+        super(SLinux, self).__init__(programs,
+                                     argv=argv,
+                                     envp=envp,
+                                     disasm=disasm)
+
 
     def _mk_proc(self, arch):
         if arch in {'i386', 'armv7'}:
             mem = SMemory32(self.constraints)
         else:
             mem = SMemory64(self.constraints)
-        return CpuFactory.get_cpu(mem, arch)
+
+        if is_binja_disassembler(self.disasm):
+            return self._init_binja_cpu(mem)
+
+        cpu = CpuFactory.get_cpu(mem, arch)
+
+        # FIXME
+        arch_map = {
+            'i386': (cs.CS_ARCH_X86, cs.CS_MODE_32),
+            'amd64': (cs.CS_ARCH_X86, cs.CS_MODE_64),
+            'armv7': (cs.CS_ARCH_ARM, cs.CS_MODE_ARM)
+        }
+        arch, mode = arch_map[arch]
+        cpu.__class__.disasm = init_disassembler(self.disasm,
+                                                 arch,
+                                                 mode,
+                                                 self.programs)
+        return cpu
+
+    def _init_binja_cpu(self, memory):
```

Comment:
That is not a problem because we are no longer relying on the `@instruction` decorator to discover code. Even if you start executing at the start of the packer, you will properly determine what to execute next based on the current instruction.